aten/src/ATen/record_function.h

```diff
@@ -1,12 +1,18 @@
 #pragma once
 
 #include <ATen/core/ivalue.h>
-#include <c10/util/SmallVector.h>
+#include <ATen/core/operator_name.h>
 #include <c10/macros/Export.h>
+#include <c10/util/Optional.h>
+#include <c10/util/SmallVector.h>
 #include <memory>
 
 #include <functional>
 
+namespace c10 {
+class CAFFE2_API OperatorHandle;
+}
```

Comment:
The overload below `void before(c10::OperatorHandle const& op, int64_t sequence_nr = -1);` takes in `c10:OperatorHandle const&`, and the class doesn't seem to be forward declared. Also I think OperatorHandle itself is declared in Dispatcher.h, so including it would cause a circular dependency.